LiveSplit.ActionHenk.asl

```diff
@@ -1,286 +1,270 @@
-    if(vars.old_sum_of_medals < 38) return vars.new_sum_of_medals >= 38;
-    if(vars.old_sum_of_medals < 43) return vars.new_sum_of_medals >= 43;
-    if(vars.old_sum_of_medals < 57) return vars.new_sum_of_medals >= 57;
-    if(vars.old_sum_of_medals < 72) return vars.new_sum_of_medals >= 72;
-    if(vars.old_sum_of_medals < 85) return vars.new_sum_of_medals >= 85;
-    if(vars.old_sum_of_medals >= 85 &&
-       current.kentinator == 1 &&
-       old.kentinator == 0) return true;
-
-    // The run is finished when the player has beaten Kentinator and finished
-    // playing the Credits level. The split should only occur when on the post
-    // game screen. Otherwise, starting splits on the main menu without clearing
-    // medal data will immediately run through all of the splits.
-    if( current.kentinator == 1 &&
-        current.last_level == vars.credits_level_id &&
-        current.game_state == vars.state_post_game) return true;
-  }
-  else if(settings["category_all_levels"]) {
-    // Splits are based on the number of batches marked as complete
-    return vars.new_batches_completed == vars.old_batches_completed + 1;
-  }
-  else if(settings["category_all_rainbows"]) {
-    return vars.new_perfect_batches == vars.old_perfect_batches + 1;
-  }
-  else if(settings["category_hundo"]) {
-    return vars.new_full_batches == vars.old_full_batches + 1;
-  }
-
-  // If none of the above conditions have been met, do not split.
-  return false;
+        if((ptrActionHenkStart = scanner.Scan(vars.scanActionHenkStart)) != IntPtr.Zero) {
+            print("[Autosplitter] ActionHenk:Start Found : " + ptrActionHenkStart.ToString("X"));
+            break;
+        }
+    }
+
+    if (ptrActionHenkStart == IntPtr.Zero)
+        throw new Exception("[Autosplitter] Can't find signature");
+
+    int relPtr = (int)((long)ptrActionHenkStart - (long)modules.First().BaseAddress);
+
+    vars.watchers = new MemoryWatcherList() {
+        // LevelBatchManager
+        // Number of medal earned in the current level
+        (vars.bestMedal =         new MemoryWatcher<int>(new DeepPointer((int)relPtr+0x472, 0x24, 0x4, 0x0, 0x2C, 0x34))),
+        // Id/code of the current level
+        (vars.levelCode =         new MemoryWatcher<int>(new DeepPointer((int)relPtr+0x472, 0x24, 0x4, 0x0, 0x2C, 0x74))),
+        // Total number of medals while in the main menu. Otherwise total number of rainbow medals
+        (vars.numMedals =         new MemoryWatcher<int>(new DeepPointer((int)relPtr+0x472, 0x24, 0x4, 0x0, 0x38))),
+        // Id of the current batch
+        (vars.lookingAtBatchNum = new MemoryWatcher<int>(new DeepPointer((int)relPtr+0x472, 0x24, 0x4, 0x0, 0x40))),
+
+        // GUIManager
+        // The active GUI screen being displayed
+        (vars.activeScreen = new MemoryWatcher<int>(new DeepPointer((int)relPtr+0x568, 0x24, 0x4, 0x0, 0x20)))
+    };
+
+    vars.UpdateResetTracker();
+
+    refreshRate = 200/3d;
```

Comment:
what does this actually change? How frequent does this end up running?

Reply:
It's the default refresh speed in Livesplit.
I re-assign it when the signature scan is done since I modify it at a speed of half a second at the very beginning of startup. I do this since signature scanning is quite cpu intensive and it's a security to not check more than needed.
Though it's not really important since the signature is loaded quite fast when the game is booted but it's better in case the signature isn't found. Just a good habit to have.